cmd/kubeadm/app/cmd/phases/upgrade/node/kubeletconfig.go

```diff
@@ -71,7 +71,17 @@ func runKubeletConfigPhase() func(c workflow.RunData) error {
 			return err
 		}
 
-		// TODO: Checkpoint the current configuration first so that if something goes wrong it can be recovered
+		// Create a copy of the kubelet config file in the /etc/kubernetes/tmp/ folder.
+		backupDir, err := constants.CreateTempDirForKubeadm(constants.KubernetesDir, "kubeadm-kubelet-config")
+		if err != nil {
+			return err
+		}
+		src := filepath.Join(kubeletDir, constants.KubeletConfigurationFileName)
+		dest := filepath.Join(backupDir, constants.KubeletConfigurationFileName)
+		if err := os.Rename(src, dest); err != nil {
+			return err
+		}
+		fmt.Printf("[upgrade] Kubelet config file is backed up to the %s. \n", backupDir)
 
```

Comment:
since we have the `--cleanup-tmp-dir` option for `kubeadm reset`, the new folder will be taken care by that with other temp folders, so I think it's fine to not clean this folder for this command alone.